doctr/models/_utils.py

```diff
@@ -75,10 +98,20 @@ def estimate_orientation(img: np.ndarray, n_ct: int = 50, ratio_threshold_for_li
             angles.append(angle - 90)
 
     if len(angles) == 0:
-        return 0  # in case no angles is found
+        estimated_angle = 0  # in case no angles is found
     else:
         median = -median_low(angles)
-        return round(median) if abs(median) != 0 else 0
+        estimated_angle = -round(median) if abs(median) != 0 else 0
+
+    # combine with the general orientation and the estimated angle
+    if page_orientation and orientation_confidence >= min_confidence:
+        # special case where the estimated angle is mostly wrong:
+        # when estimated angle is + or - 90 degree, we should consider the general orientation
+        if abs(estimated_angle) == 90 and abs(page_orientation) in [0, 90]:
+            return page_orientation
```

Comment:
I don't understand entirely this section. Here, `estimated_angle == 90`, so according to your comment, it's wrong, so `page_orientation` is returned but it's also `90`. It means there is no evident fallback to fix this issue ?

Reply:
yeah the problem case is if estimated_angle is exact 90 or -90 it's mostly not correct (swapping - and + or complete wrong)

So we should check the page orientation also..

I think we can simplify this:

```
if abs(estimated_angle) == abs(page_orientation):
    return page_orientation
```

wdyt ?

Reply:
But what "mostly wrong" means here ? Based on your experimentations, if it's 90 or -90, it's completly wrong ?

Reply:
In the most cases yes

Reply:
@odulcy-mindee 

pushed a small update 

tested again with a complex set of documents (mobile phone, receipts, etc) 30 samples

randomly rotated in range -180 - 180

it mostly varies between 22 - 25 correct from 30 

before this integration / changes on the same set we have had 7 - 11 correct so as mentioned still some space for further improvements it's not perfect but already much better as before :)

Reply:
ah.. if i lower the min_confidence we get ~27 - 29 correct xD trust the page orientation model xD

Reply:
ok, it sounds ok for a "first" iteration. We can improve later in another PR

Reply:
yep i think the same